countdown_timer.jsx

```diff
@@ -91,6 +93,7 @@ var CountdownTimer = React.createClass({
 
     if (countdownComplete) {
       if (this.props.completeCallback) { this.props.completeCallback() };
+      if(this.props.shouldReplayTimer){timeRemaining=this.props.initialTimeRemaining};
```

Comment:
This won't actually update the component's state. You need to use `this.setState()` to update `this.state.timeRemaining`.

See: https://facebook.github.io/react/docs/tutorial.html#reactive-state